### PR TITLE
fix warnings in torpedo store

### PR DESCRIPTION
**Pull request by sebestyn**
please check my fixes

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -13,6 +13,8 @@ public class TorpedoStore {
   private double FAILURE_RATE = 0.0; //NOSONAR
 
   private int torpedoCount = 0;
+  private Random generator = new Random();
```
Comment (krafcsikgergo) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:16]: Great idea.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (krafcsikgergo) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: sooo true.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
...
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (krafcsikgergo) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:45]: silly mistake.

**Review by krafcsikgergo — CHANGES_REQUESTED** [review]
Nicely done!

**Review by krafcsikgergo — APPROVED** [review]
Nice!!!!!!!!

**Comment by sebestyn** [comment]
minden javítva, köszönöm mindenkinek a munkát

**Comment by sebestyn** [comment]
reopen to merge

**Pull request merged**